<launch>
    <arg name="driver_node_name" default="roverrobotics_driver" />
    <arg name="config_locks" default="$(find roverrobotics_driver)/config/twist_mux_locks.yaml" />
    <arg name="config_topics" default="$(find roverrobotics_driver)/config/twist_mux_topics_locks.yaml" />
    <param name="device_port" value="/dev/rover-pro" />
    <param name="comm_type" value="serial" />
    <param name="robot_type" value="pro" />
    <param name="robot_mode" value="TRACTION_CONTROL" />
    <param name="status_frequency" value="30" />
    <param name="Kp" value="0.4" />
    <param name="Ki" value="0.7" />
    
    
    <!-- RoverRobotics Ros 1 Driver -->
    <node 
        pkg="roverrobotics_driver" 
        type="roverrobotics_driver_node" 
        name="$(arg driver_node_name)" 
        respawn="true"
        repsnaw_delay="30"
        output="screen">
    </node>

    <!--  Twist Mux  -->
    <node pkg="twist_mux" type="twist_mux" name="twist_mux" output="screen">
        <remap from="cmd_vel_out" to="/cmd_vel/managed" />

        <rosparam file="$(arg config_locks)" command="load" />
        <rosparam file="$(arg config_topics)" command="load" />
    </node>

    <!-- Control input manager -->
    <node pkg="roverrobotics_input_manager" type="control_input_manager.py" name="roverrobotics_input_manager_node" output="log">
        <rosparam file="$(find roverrobotics_input_manager)/config/input_topics.yaml" command="load" />
    </node>

</launch>
